### DISCUSSION TITLE
Configuring simple modifications

### DISCUSSION THREAD

**Discussion by hadynz** [evan-liu/karabiner.ts#104] [Q&A] [answered]
My current keyboard requires me to swap my left `option` and `command` with each other. 

So I did the following and got things working:

```
rule('...').manipulators([
  map('left_command').to('left_option'),
  map('left_option').to('left_command'),
]);
```

However, the problem with the above configuration is that it implements this mapping with a "complex modification". This makes me lose repeat keys, and prevents me from using shortcuts that trigger when both modifiers (`option` + `command`) are pressed together.

Is there a way to use `karabiner.ts` and to configure a "simple modification"?

**Comment by evan-liu** [comment, marked as answer]
Can you try to see if this solves your problems? 

```typescript
map('left_command', 'optionalAny').to('left_option'),
map('left_option', 'optionalAny').to('left_command'),

// or
map('‹⌘', '??').to('‹⌃'),
map('‹⌃', '??').to('‹⌘'),
```

Karabiner-Elements has an easy to use UI for "simple modification", and `karabiner.ts` should be able to work with it without problems. Please let me know if that's not the case.

**Reply by hadynz** [reply]
That worked really well. Might be worth highlighting in the documentation.

> Karabiner-Elements has an easy to use UI for "simple modification", 

I think adding support for this would be a good win. I see many people like myself who want ALL of their configuration stored in code for consistency across development environments.
